### ISSUE TITLE
[3.x] Several minor LSP  issues

### ISSUE THREAD

**Issue by Rubonnek** [godotengine/godot#53293]
### Godot version

3.x branch at 8d0b2edfc25dec38fff36e4b198753e570cd4ec9

### System information

Arch Linux

### Issue description

This ticket is following up on https://github.com/godotengine/godot/issues/53238#issuecomment-931225975. On that issue, #53238, it was determined that one of the underlying causes that made the engine to crash was a buggy LSP client.

For a while now I've been stumbling upon strange errors ~that seem to be caused by a buggy LSP client, in this case Vim-ALE,~ and I'm bringing this up in case there are relevant changes that should be made to Godot.

Most of my tests are on 3.x. I am unsure if the issues below affect the `master` branch.

The errors described below are also not shown in the editor, but in an external terminal emulator that is attached to Godot.

### Steps to reproduce

All the cases below assume the user has opened the engine using `godot -e` at the path where `project.godot` is at.

## Case 1

Hovering over the `extends` in a GDScript file generates the following LSP event:
```json
{"method": "textDocument/hover", "jsonrpc": "2.0", "id": 2, "params": {"textDocument": {"uri": "file:///home/wilson/game_development/godot/godot-template/src/addons/tiled_importer/custom_property_manager_for_tilesets_template.gd"}, "position": {"character": 0, "line": 1}}}
```
which generates the following Error:
```
ERROR: Method failed.
   at: _determine_inheritance (modules/gdscript/gdscript_parser.cpp:5423)
```
## Case 2
Hovering over an empty line generates the following LSP event:
```json
{"method": "textDocument/hover", "jsonrpc": "2.0", "id": 2, "params": {"textDocument": {"uri": "file:///home/wilson/game_development/godot/godot-template/src/addons/tiled_importer/custom_property_manager_for_tilesets_template.gd"}, "position": {"character": 0, "line": 1}}}
```
Which generates the following error:
```
ERROR: Index p_position.character = 0 is out of bounds (line.size() = 0).
   at: get_identifier_under_position (modules/gdscript/language_server/gdscript_extend_parser.cpp:490)
ERROR: Index p_position.character = 0 is out of bounds (line.size() = 0).
   at: get_identifier_under_position (modules/gdscript/language_server/gdscript_extend_parser.cpp:490)
```
## Case 3
Opening a GDScript file outside of the Godot project generates a ton of events, but the following seems to be the important one:
```
{"method": "textDocument/didOpen", "jsonrpc": "2.0", "params": {"textDocument": {"uri": "file:///tmp/outside_of_any_project/Node2D.gd", "version": 1, "languageId": "gdscript3", "text": "extends Node2D\n"}}}
```
And right after you'll see the following error:
```
ERROR: Condition "err" is true. Returned: err
   at: load_source_code (modules/gdscript/gdscript.cpp:775)
ERROR: Cannot load source code from file 'file:/tmp/outside_of_any_project/Node2D.gd'.
   at: load (modules/gdscript/gdscript.cpp:2177)
ERROR: Failed loading resource: file:/tmp/outside_of_any_project/Node2D.gd. Make sure resources have been imported by opening the project in the editor at least once.
```

### Minimal reproduction project

A GDScript template will do:

```gdscript
extends Node2D

func _ready() -> void:
	pass
```

**Comment by Rubonnek** [comment]
CC @Razoric480

**Comment by Rubonnek** [comment]
I've tested the steps above with VSCode in Linux and it doesn't seem like these minor issues are caused by a buggy LSP client this time.

**Comment by Razoric480** [comment]
Two PRs have been submitted to fix the issue - 2 and 3 exist on 4.0 and so should be cherry-picked in #53308, and 1 only exists in 3.x and is fixed by #53309.

**Comment by akien-mga** [comment]
Fixed by #53308 and #53309.
